### ISSUE TITLE
I can't get @yieldparam to work

### ISSUE THREAD

**Issue by Hubro** [castwide/solargraph#159]
I'm using VSCode with Solargraph 0.31.2 and I can't get Solargraph to interpret the type of my objects in a block based on `@yieldparam`.

Here's a function:

```ruby
##
# Reads a stream and yields every object
#
# @yieldparam [String] name
# @yieldparam [Hash] object
#
def read(stream)
  lines = enumerate(stream)

  read_objects(lines) { |name, object|
    yield name, object
  }
end
```

When I use it like this:

```ruby
MyModule.read(io) { |name, object|
  # At this point, Solargraph has no idea about the types of "name" and "object"
}
```

I know Solargraph is set up correctly, because when I hover the "read" function it shows the documentation for it, and I can jump to the definition.

If I add a couple of `@param` lines like this:

```ruby
# @param [String] name
# @param [Hash] object
MyModule.read(io) { |name, object|
  # Now Solargraph knows the types of "name" and "object" and correctly suggests methods
}
```

I would expect (and I've read) that Solargraph can infer the types of the parameters by the `@yieldparam` documentation of the yielding function. Why isn't it working in this case? Am I doing something wrong?

**Comment by castwide** [comment]
This is an odd one. It looks like the `##` at the beginning of the documentation is the problem. I was able to get the correct behavior by removing them:

```ruby
module MyModule
  # Reads a stream and yields every object
  #
  # @yieldparam [String] name
  # @yieldparam [Hash] object
  #
  def self.read_something(stream)
    lines = enumerate(stream)

    read_objects(lines) { |name, object|
      yield name, object
    }
  end
end

MyModule.read_something(io) do |name, object|
  name. # <- suggests String methods
end
```

I'll look further into it and try to have a fix for the next patch release.

**Comment by Hubro** [comment]
> ...It looks like the `##` at the beginning of the documentation is the problem.

Can confirm. I've removed those lines from my comments and it works just as expected now. Additionally, the documentation showed when hovering classes and methods with the mouse is now properly formatted. That's probably related.

**Comment by castwide** [comment]
Yep, they're definitely related. There's a fix in the master branch. I'll include it in the next release.

**Comment by Hubro** [comment]
Awesome! Thanks for the quick fix.

**Comment by castwide** [comment]
Fix released in v0.32.0.
